doom.go

```diff
@@ -19,6 +20,36 @@ import (
 	"unsafe"
 )
 
+var vfs fs.StatFS
+
+func init() {
+	vfs = os.DirFS(".").(fs.StatFS)
+}
+
+// VFSAdapter adapts an fs for statfs
```

Comment:
I don't quite understand this - there's no users of VFSAdapter?

Reply:
Oh, I think I see. You're using it to allow `embed.FS` to work externally? Hmm... I wonder if we can get rid of the calls to `Stat` to make this simpler.

Reply:
ok

Reply:
When I made the first comment about unifying the two implementations, I thought that `embed.FS` implemented `fs.StatFS`, but it doesn't.